services/auth/authServices.ts

```diff
@@ -0,0 +1,27 @@
+import { AxiosResponse } from "axios";
+import api from "@shared/fetcher";
+import { User } from "@services/initChatAnonymous/initChatAnonymousServices";
+
+type CheckExchangeTokenResponse = {
+  code: number;
+  data: boolean;
+  message: string;
+  token: string;
+  anonymousToken: string;
+  refresh_token: string;
+};
+
+type CheckExchangeTokenPayload = { exchangeToken: string };
+
+const checkExchangeToken = async (
+  payload: CheckExchangeTokenPayload
+): Promise<AxiosResponse<User> & CheckExchangeTokenResponse> => {
+  const { data } = await api.post<AxiosResponse<User> & CheckExchangeTokenResponse>(
+    "/api/v1/users/authenticate-web",
+    payload
+  );
+  return data;
+};
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The return type of the `checkExchangeToken` function seems to be incorrect. It's currently defined as `Promise<AxiosResponse<User> & CheckExchangeTokenResponse>`, which implies that the returned promise will resolve with an object that is both an `AxiosResponse<User>` and a `CheckExchangeTokenResponse`. However, the function implementation suggests that it should only return a `CheckExchangeTokenResponse`.

```diff
- ): Promise<AxiosResponse<User> & CheckExchangeTokenResponse> => {
+ ): Promise<CheckExchangeTokenResponse> => {
```
Also, you are returning `data` directly from the Axios response. This might lead to unexpected results if the API call fails but doesn't throw an error (for example, if it returns a 404 or 500 status). You should add error handling here to ensure that any API errors are properly caught and handled.

```diff
  const { data } = await api.post<AxiosResponse<User> & CheckExchangeTokenResponse>(
    "/api/v1/users/authenticate-web",
    payload
  );
- 23:   return data;
+ 23:   if (data.code === 200) {
+ 24:     return data;
+ 25:   } else {
+ 26:     throw new Error(data.message);
+ 27:   }
```
This way, if the API call returns a non-200 status code, an error will be thrown with the message from the API response.


<!-- This is an auto-generated comment by OSS CodeRabbit -->